### ISSUE TITLE
[FEA] Pass `cudaStreamPerThread` to numba/CuPy kernels

### ISSUE THREAD

**Issue by shwina** [rapidsai/cudf#5922]
Memory allocations should already use PTDS since both numba and CuPy allocate memory using RMM. Kernels on the other hand, may explicitly need to be passed the `cudaStreamPerThread` stream handle.

cc: @jakirkham @kkraus14

**Comment by jakirkham** [comment]
cc @pentschev (for awareness)

**Comment by vyasr** [comment]
@shwina Is this intended to be unconditional, or only when requested by the user? We will also need to start passing streams down to libcudf everywhere in order to support this.

**Comment by pentschev** [comment]
My take on this is that we could begin with an opt-in solution, like we've done for CuPy with [`CUPY_CUDA_PER_THREAD_DEFAULT_STREAM`](https://docs.cupy.dev/en/stable/reference/environment.html#envvar-CUPY_CUDA_PER_THREAD_DEFAULT_STREAM). This could allow us to test things without being too intrusive, and if this proves to be useful then we could enable it by default.

**Comment by jakirkham** [comment]
Yeah agreed. We probably want to support both the current mode and PTDS for some time. There are likely different issues we will run into. So having an escape hatch to old behavior is quite useful.
